x-pack/plugins/lists/common/schemas/types/non_empty_entries_array.ts

```diff
@@ -21,6 +22,14 @@ export const nonEmptyEntriesArray = new t.Type<EntriesArray, EntriesArray, unkno
     if (Array.isArray(input) && input.length === 0) {
       return t.failure(input, context);
     } else {
+      if (
+        Array.isArray(input) &&
+        input.some((entry) => entriesList.is(entry)) &&
+        input.some((entry) => !entriesList.is(entry))
```

Comment:
Could we make arrays of list entries and arrays of non-list entries into separate types? Then `entriesArray` would be the union of those types and we avoid writing custom validation logic.

Reply:
I'm not sure there is a way to create an XOR with io-ts so the best we can do is custom validation.

Reply:
I just happened to see this float by, so forgive me if I'm misunderstanding... but I think this should be possible... something like `entries = t.union[t.array(listEntries), t.array(otherEntries)]`, right?

Reply:
Yeah @madirey I think that would work

Reply:
Awesome thanks everyone! That seems to have worked 👍

Reply:
Discussed with @madirey @marshallmain - since this is a temporary stop-gap for a functional issue and not an issue with the schema, we will be utilizing the custom validation logic. The issue that arose with replacing `entriesArray` with the unioned type revolved around iterating over objects of type `entriesArray`. Typescript won't let you iterate over a unioned type since a union type is not a valid iterable signature. Thus we would need explicit type guards in every place we iterate over objects of type `entriesArray` if we moved forward with the unioned type. Since this is a temporary fix for something we plan to change in the future to accurately represent the functionality the schemas themselves reflect, the custom validation logic will be the quickest to remove and the less invasive going forward.

Reply:
^^^ Just seconding this, we're good to move forward with custom validation.